### PR TITLE
Migrate Enter OTP screen to Mobius: Stabilize the tests

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/502/migrate-enter-otp-screen-to-mobius

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt [hunk 1]
```diff
@@ -113,57 +111,45 @@ class EnterOtpScreenControllerTest {
...
 
   @Test
-  @Parameters(
-      "|false",
-      "1|false",
-      "11|false",
-      "111|false",
-      "1111|false",
-      "11111|false",
-      "111111|true"
```
Comment (Honey14) [inline app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt:151]: Removing parameters will also remove multiple test cases, right?

File: app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt [hunk 2]
```diff
@@ -387,100 +390,166 @@ class EnterOtpScreenControllerTest {
...
-    )
+    uiEvents.onNext(EnterOtpResendSmsClicked())
+
+    verify(screen, never()).showSmsSentMessage()
+  }
+
+  @Test
+  fun `when the resend sms call fails with a server error, the sms sent message must not be shown`() {
+    whenever(activateUser.activate(loggedInUserUuid, pin))
+        .doReturn(ActivateUser.Result.ServerError(400))
```
Comment (Honey14) [inline app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt:731]: Should the `ServerError` code be 500 instead or 400 is fine?

File: app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt:151]: Yeah, this was intentional. I felt the overhead of having parameterized tests was not worth the effort of testing each edge case here.

File: app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt:731]: Good point; Although it makes no difference to the tests, we should strive to match real world scenarios. I'll make this change in a later PR.

File: app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt (hunk 1 above)
Comment (Honey14) [reply app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt:151]: So we test the success case and failed case only, right?
